6_visualize/src/prep_spark_line_fun.R

```diff
@@ -0,0 +1,71 @@
+
+prep_spark_line_fun <- function(storm_data_ind_file, viz_config_yaml, DateTime){
+  storm_data <- readRDS(sc_retrieve(storm_data_ind_file))
+  viz_config <- yaml::yaml.load_file(viz_config_yaml)
+
+  sites <- unique(storm_data$site_no)
+  this_DateTime <- as.POSIXct(DateTime, tz = "UTC") # WARNING, IF WE EVER MOVE FROM UTC elsewhere, this will be fragile/bad.
+  this_spark <- filter(storm_data, dateTime <= this_DateTime) # keep all data up until this timestep
+
+  plot_fun <- function(){
+    orig_par <- par()
+
+    # Set spacing configurations
+    x_coords <- c(viz_config$sparks$xleft, viz_config$sparks$xright)
+    y_start <- viz_config$sparks$ybottom
+    y_space <- viz_config$sparks$ytop-y_start
+    vertical_spacing <- y_space/length(sites)
+
+    y_pos <- y_start # initialize position of first spark
+    for(i in sites) {
+
+      # Filter data to just one site & create polygon out of it
+      storm_data_i <- filter(this_spark, site_no == i) %>% arrange(dateTime)
+      storm_data_site_all <- filter(storm_data, site_no == i)
+
+      if(nrow(storm_data_i) == 0) { next } # skip if there's no data at or before this timestep
+
+      flood_stage_va <- as.numeric(unique(storm_data_i$flood_stage_normalized))
+      lowest_stage <- min(storm_data_site_all$stage_normalized, na.rm = TRUE)
```

Comment:
shouldn't lowest_stage always be 0 for normalized stage? to check, i ran
```r
> storm_data %>% group_by(site_no) %>% summarize(min=min(stage_normalized, na.rm=TRUE), max=max(stage_normalized, na.rm=TRUE))
```
and got
```
# A tibble: 20 x 3
   site_no    min   max
   <chr>    <dbl> <dbl>
 1 08033500    0.    1.
 2 08040600    0.    1.
 3 08041000    0.    1.
 4 08041780    0.    1.
 5 08066250    0.    1.
 6 08066500    0.    1.
 7 08067000    0.    1.
 8 08072050    0.    1.
 9 08106350    0.    1.
10 08106500    0.    1.
11 08111500    0.    1.
12 08114000    0.    1.
13 08116650    0.    1.
14 08159500    0.    1.
15 08160400    0.    1.
16 08161000    0.    1.
17 08162000    0.    1.
18 08162500    0.    1.
19 08173900    0.    1.
20 08175800    0.    1.
```

Reply:
I didn't think we wanted to hard code anything in case we changed how we normalized them?

Reply:
I think we can and should assume normalized means between 0 and 1 - it adds a lot of code to this function to avoid that assumption, and it's one we have control over in the 2_process phase. You could add a comment to 2_process/src/normalize_streamdata.R making that connection explicit, if you are worried about it.

Reply:
Ok, I see what you mean. I can do that!

Reply:
I actually just realized this should not have been working perfectly to begin with. The stage is normalized between 0 and 1, but the flood stage could be above that.
```
> l <- readRDS("2_process/out/normalized_streamdata.rds")
> summary(l$stage_normalized)
   Min. 1st Qu.  Median    Mean 3rd Qu.    Max.    NA's 
0.00000 0.07856 0.52283 0.49679 0.91690 1.00000     224 
> summary(l$flood_stage_normalized)
   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.1233  0.4785  0.6223  0.6577  0.7756  1.2874 
```
I could change the limits to `ylim=c(0, flood_stage_va)` ?

Reply:
ooh, good observation...but i think it's reasonable to set the ylims to the normalized data (0 to 1) rather than to compute them from the theoretical (the flood stage), because some streams might stay well below that and others could go well above it, and we'd still want to be able to see the full variation in each type of stream

Reply:
Hmmm, but then will we ever be able to see the flood stage? Or would it get cut off? It is somehow not right now and I can't seem to wrap my head around why

Reply:
We would see flood stage if and only if the actual stage ever got as high as flood stage. i think that's enough, isn't it?

What are you seeing that makes you say it's not right now?

Reply:
OH GOOD GRIEF - DUH! Yep, I got it now :) If a site hits flood stage, then that flood stage is included in the 0 to 1 already :)

Reply:
:smile: